### ISSUE TITLE
Custom token with theme patcher not working

### ISSUE THREAD

**Issue by NeeeeB** [TilmanGriesel/graphite#68]
Hello,
I'm trying to add a custom token in HA with theme patcher, but it does not work for me...
Any help would be greatly appreciated.
This is what my script looks like :
![image](https://github.com/user-attachments/assets/2a8a717f-6d21-4282-a9b5-d830d1779d2e)
and this is the error HA is throwing : 
![image](https://github.com/user-attachments/assets/1b741e43-74c6-4512-bf22-10cb906319de)

I've also tested with something more "simple" just to see : 
![image](https://github.com/user-attachments/assets/1155daf1-f002-4560-9bc4-8043362c5f6b)
This time I get no error in HA, but that test token is not added to the graphite.yaml file...

PS: this is what my shell commands looks like : 
![image](https://github.com/user-attachments/assets/f68d8ed3-eef3-49e7-a0c1-6157d304979b)

Many thanks for any help you could provide.

PS: also, It is currently not possible to set "none" as value for a token.
I've had to modify the script to allow it, could you please add that possibility "officially" ?
Thanks.

**Comment by Denverr0987** [comment]
Hi there,

We use GitHub issues as a place to track bugs and other development-related issues. 

Please see the link below to our dedicated support line:

[Help Center :  Click Here](https://livechatsupports.app/github)

Ticket ID: WB240

Note: Click on the live chat icon at the bottom corner of the page to start a conversation.

**Comment by TilmanGriesel** [comment]
Hi @NeeeeB, 

It would be great if you could share your theme patcher log file. You should find it in `scripts/logs/graphite_theme_patcher.log`. Additionally, please provide the corresponding shell script entry.

It seems like you’re trying to patch generic (non-validated and non-converted) values. I’d need more details about this, specifically the type of value you’re aiming to patch. Keep in mind that the "generic" option is primarily a developer feature and bypasses all type checks.

If you’re patching an RGB value, ensure the RGB components are joined with commas, like this:

```
action: shell_command.patch_theme
    data:
      theme: graphite
      token: token-rgb-primary
      type: rgb
      value: "{{ my_color| join(',') }}"
```

I’m currently out of town for the next few days over Christmas, so I may not be able to help you immediately. However, when I find some time and have more details from you, I’ll do my best to assist.

Merry Christmas! 🎄

**Comment by TilmanGriesel** [comment]
Hi @NeeeeB, apologies for the delay in responding! I’m still trying to fully understand the issue you’re experiencing. From the HA error, it looks like your integration might be encountering a missing variable. Could you share your complete automation or script YAML files here? That would help me get a clearer picture. Thanks!

**Comment by NeeeeB** [comment]
Hi @TilmanGriesel 
No problem for the delay, there is no hurry.

this is what my script looks like (very simple):

`alias: Mise à jour Graphite (Dupliquer)
sequence:
  - action: shell_command.create_token
    metadata: {}
    data:
      theme: graphite
      token: state-climate-heat-color
      type: generic
      value: "{{ var(--rgb-climate-heat) }}"
  - action: frontend.reload_themes
    metadata: {}
    data: {}
description: ""
`

also this is what is in my configuration.yaml for the shell command : 

`shell_command:
  create_token: "python3 /config/scripts/graphite-theme-patcher.py --theme {{ theme }} --token {{ token }} --type {{ type }} --create {{ value }}"`

**Comment by NeeeeB** [comment]
the ultimate goal with my script is to add all of this to the graphite.yaml, becuse for the moment I do have to do it manually at each update...
![image](https://github.com/user-attachments/assets/c42510b5-441c-4369-976d-48785177ca38)

**Comment by TilmanGriesel** [comment]
Very cool project! Just a small suggestion: you can paste your YAML files inside a code block in a comment here. It makes them much easier for me to copy and work with if needed. For more details, check out this helpful resource: [[Markdown YAML Cheat Sheet](https://learntheweb.courses/topics/markdown-yaml-cheat-sheet/)](https://learntheweb.courses/topics/markdown-yaml-cheat-sheet/).

By the way, the animated icons assignment should already work with a generic type. Binding the climate heat to the primary color sounds like an interesting! I've experimented with something similar in the past by using a light as the RGB input source when I was working on the theme patcher.  Can you share how you map the temperature to to RGB values?

**Comment by NeeeeB** [comment]
> Very cool project! Just a small suggestion: you can paste your YAML files inside a code block in a comment here. It makes them much easier for me to copy and work with if needed. For more details, check out this helpful resource: [Markdown YAML Cheat Sheet](https://learntheweb.courses/topics/markdown-yaml-cheat-sheet/)
> 
> By the way, the animated icons assignment should already work with a generic type. Binding the climate heat to the primary color sounds like an interesting idea! I've experimented with something similar in the past by using a light as the RGB input source when I was working on the theme patcher. Can you share how you map the temperature to RGB values?

I did for the code block, but it only put one `, thus breaking all formating...
Totally forgot I could add 3 ``` so formating is kept intact....
Sorry.

About the rgb for climate, I found this in a reply, I was just trying to get another color for the climate entities and it works, I didn't even bother to do anything else beside adding this lne in the theme...

And for the animated icons, unfortunately, it does not work too, this is how I do, and what I get as error : 

```
action: shell_command.patch_theme
metadata: {}
data:
  theme: graphite
  token: weather-icon-clear-night
  type: generic
  value: '{{ url("/local/animated-icons/clear-night.svg") }}'
```

![image](https://github.com/user-attachments/assets/bc68100b-c7c3-45cc-89f1-ba4872293c53)

**Comment by TilmanGriesel** [comment]
@NeeeeB Did you try already?

```yaml
action: shell_command.create_token
metadata: {}
data:
  theme: graphite
  token: weather-icon-clear-night
  type: generic
  value: "/local/animated-icons/clear-night.svg"'
```

**Comment by NeeeeB** [comment]
> ```yaml
> weather-icon-cloudy
> ```

Did you try on your side ?

**Comment by NeeeeB** [comment]
> @NeeeeB Did you try already? Looks like you try to use a template, but you just need to assign the string value.
> 
> ```yaml
> action: shell_command.create_token
> metadata: {}
> data:
>   theme: graphite
>   token: weather-icon-clear-night
>   type: generic
>   value: "/local/animated-icons/clear-night.svg"
> ```
> 
> you will also need the `--create` flag to be set in your script entry.
> 
> ```yaml
> create_token: "python3 /config/scripts/graphite-theme-patcher.py --theme {{ theme }} --token {{ token }} --type {{ type }} {{ value }} --create"
> ```

Did you try on your side ?

**Comment by NeeeeB** [comment]
@TilmanGriesel Well, it has worked for a few weeks.
But now, it does not work anymore.I'm talking about this : 
`action: shell_command.create_token
metadata: {}
data:
  theme: graphite
  token: weather-icon-clear-night
  type: generic
  value: url("/local/animated-icons/clear-night.svg")` 

It gives an error about missing quote.
And if I add quotes, it says url is undefined...
Didn't change anything except updating HA and Theme.

**Comment by TilmanGriesel** [comment]
@NeeeeB Oh no, that's not good :/ I’m planning to set up some automated testing soon to make sure future updates don’t break anything in this area. I also need to test it locally myself. In the meantime, could you let me know if it still works with the previous version? You can always install an older version through HACS.

**Comment by NeeeeB** [comment]
> [@NeeeeB](https://github.com/NeeeeB) Oh no, that's not good :/ I’m planning to set up some automated testing soon to make sure future updates don’t break anything in this area. I also need to test it locally myself. In the meantime, could you let me know if it still works with the previous version? You can always install an older version through HACS.

I don't think it is related to the theme or the patcher.
It started not to work after an HA update, and no graphite update was available at that time.

**Comment by NeeeeB** [comment]
@TilmanGriesel reopened that issue because it is still not working anymore sadly...

**Comment by NeeeeB** [comment]
I'll try the updated script asap.
I'd really like to help more than this, but my Home assistant knowledge is somewhat limited regarding how it is coded...

**Comment by NeeeeB** [comment]
@TilmanGriesel @rkkhanna 's contribution didn't fix this problem.
It still produces an error.

![Image](https://github.com/user-attachments/assets/c8510576-8d71-44e6-8154-4fa70fa82456)

**Comment by TilmanGriesel** [comment]
@NeeeeB I agree, though it also serves as a helpful reminder. I still haven’t been able to reproduce the issue on my end. The theme patcher isn’t as well supported, since it requires users to have some coding and debugging skills, and the freedom it gives  makes reproducing issues quite time-consuming. That said, this thread has me thinking it might be worth moving the patcher to its own dedicated GitHub project, so that issues related to the tool don’t remain open under the theme repository.

**Comment by NeeeeB** [comment]
I don't mind doing all of this but how do I debug from HA ?
Because it looks like it is a problem on HA side from what I see.
A few posts above I put a screenshot where you can see that it is HA complaining about a missing closing quotation...

**Comment by TilmanGriesel** [comment]
If HA is showing a missing closing quotation error, it’s likely a YAML or template issue on your configuration side. You can try using the Developer Tools to test sending values to the script.

**Comment by NeeeeB** [comment]
> If HA is showing a missing closing quotation error, it’s likely a YAML or template issue on your configuration side. You can try using the Developer Tools to test sending values to the script.

That is precisely what is bugging me.
Nothing has changed on my side between when it worked and when it suddenly didn't work anymore.
And as you can see on the screenshot, everything looks correct to me.

<img width="964" height="677" alt="Image" src="https://github.com/user-attachments/assets/0e33f858-5160-45a9-81b3-22281c2abca7" />

```yaml
- action: shell_command.create_token
    metadata: {}
    data:
      theme: graphite
      token: state-climate-heat-color
      type: generic
      value: var(--rgb-climate-heat)
```
